client/src/components/CatalogFilterRegion/CatalogFilterRegion.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Box, Checkbox, FormControlLabel } from '@mui/material';
+import { FilterAccordion } from '../index';
+
+const regions = ['region1', 'region2', 'region3', 'region4'];
+
+const CatalogFilterRegion = () => {
+  const [checked, setChecked] = React.useState(Array(regions.length).fill(false));
+
+  const isAllChecked = checked.every((el) => el === true);
+
+  const handleChangeAll = () => {
+    setChecked((current) => {
+      const newChecked = [...current];
+      if (!isAllChecked) {
+        newChecked.fill(true);
+      } else {
+        newChecked.fill(false);
+      }
+
+      return newChecked;
+    });
+  };
+
+
+  const handleChange = (event) => {
+    const index = regions.indexOf(event.target.id);
+    setChecked((current) => {
+      const newChecked = [...current];
+      newChecked[index] = !newChecked[index];
+      return newChecked;
+    });
+  };
+
+  const children = (
+    <Box sx={{ display: 'flex', flexDirection: 'column', ml: 3 }}>
+      {regions.map((el) => (
+        <FormControlLabel
+          key={el}
+          label={el}
+          control={<Checkbox id={el} checked={checked[regions.indexOf(el)]} onChange={handleChange} />}
+        />
+      ))}
+    </Box>
+  );
```

Comment:
not a very preferred method to declare children in such way.  very complicated and non readable.

Reply:
This method is offered by material ui.
![image](https://user-images.githubusercontent.com/107770819/201555913-28be6137-5c2a-4594-8643-a2ed209f4f3b.png)

What exactly is hard to read?